File: marketFE/src/app/models/command/orderCommand/searchOrderCommand.ts
import { GenericSearchCommand } from "../genericCommand/GenericSearchCommand";

export class SearchOrdersCommand extends GenericSearchCommand {
    id?: number;
    dataAcquistoDa?: string;
    dataAcquistoA?: string;
    totale?: number;
    customerId?: number;
    deleted!:boolean;
}

File: MarketBE/src/main/java/com/lucapolizzo/market/command/order/SearchOrderCommand.java
package com.lucapolizzo.market.command.order;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.lucapolizzo.market.command.generic.GenericSearchCommand;
import com.lucapolizzo.market.models.entities.User;
import lombok.Data;

import java.util.Date;

@Data

public class SearchOrderCommand extends GenericSearchCommand {
    private Integer id;

    @JsonFormat(pattern="dd/MM/yy")
    private Date dataAcquistoDa;

    @JsonFormat(pattern="dd/MM/yy")
    private Date dataAcquistoA;

    private Double totale;

    private Integer customerId;

    private Boolean deleted;
}
